standalone/prouter/src/server.rs

```diff
@@ -0,0 +1,134 @@
+use anyhow::{anyhow, Context, Error, Result};
+#[allow(unused_imports)]
+use log::{debug, error, info, warn};
+
+use rocket::config::Limits;
+use rocket::http::Method;
+use rocket_contrib::json::Json;
+use rocket_cors::{AllowedHeaders, AllowedMethods, AllowedOrigins, CorsOptions};
+use std::env;
+use std::io::Read;
+use std::sync::{Arc, Mutex};
+
+use crate::translator;
+use crate::types;
+
+use phaxt::{ParachainApi, RelaychainApi};
+
+fn cors_options() -> CorsOptions {
+    let allowed_origins = AllowedOrigins::all();
+    let allowed_methods: AllowedMethods = vec![Method::Get, Method::Post]
+        .into_iter()
+        .map(From::from)
+        .collect();
+
+    // You can also deserialize this
+    rocket_cors::CorsOptions {
+        allowed_origins,
+        allowed_methods,
+        allowed_headers: AllowedHeaders::all(),
+        allow_credentials: true,
+        ..Default::default()
+    }
+}
+
+fn send_data(
+    target_endpoint: &Vec<u8>,
+    path: &String,
+    method: &types::PRouterRequestMethod,
+    data: &Vec<u8>,
+) -> Result<Vec<u8>> {
+    let mut endpoint_str = String::from_utf8_lossy(target_endpoint).into_owned();
+
+    let mut client_builder = reqwest::blocking::Client::builder();
+    if endpoint_str.contains("b32.i2p") {
+        // a i2pd address, http is enough since i2p will encrypt msg anyway
+        endpoint_str = format!("http://{}", endpoint_str);
+        client_builder = client_builder.proxy(reqwest::Proxy::http("http://localhost:4444/")?)
+    }
+    let client = client_builder.build()?;
+
+    let target_url = format!("{}{}", endpoint_str, path);
+    debug!(
+        "PRouter send data to {}, method: {:?}",
+        &target_url, &method
+    );
+    let mut res = match method {
+        types::PRouterRequestMethod::GET => client.get(target_url).send()?,
+        types::PRouterRequestMethod::POST => client.post(target_url).body(data.clone()).send()?,
+    };
+
+    if res.status().is_success() {
+        let mut msg = Vec::new();
+        res.read_to_end(&mut msg)?;
+        Ok(msg)
+    } else {
+        Err(anyhow::Error::msg(res.status()))
+    }
+}
+
+#[post("/json/send_data", format = "json", data = "<prouter_send_json>")]
+fn json_send_data(
```

Comment:
Is this for testing purpose?

Reply:
I was thinking to let pRouter handle all routing tasks, so this is intended to be served as an endpoint for pRuntime to post. When pRuntime (or other components) wants to send a data to another worker, it can send the data by making a post request to this endpoint, without touching routing tables itself.